### ISSUE TITLE
[Bug]: defer/Await/Suspense does not show loading/fallback if re-running loader with defer

### ISSUE THREAD

**Issue by longzheng** [remix-run/react-router#10816]
### What version of React Router are you using?

6.15.0

### Steps to Reproduce

https://stackblitz.com/edit/github-osaenk?file=src%2Fapp.tsx

I've set up a very simple project that has one loader that returns `defer` with a promise that returns after a 2 second delay.

I'm using `<React.Suspense>` and `<Await>` to show a loading/fallback state when the deferred data is loading.

1. Click "Update time" button
2. Wait for text to update

### Expected Behavior

After clicking "Update time" button, I expect to see the "Loading..." text from `<React.Suspense>` for 2 seconds because the `<Await>` should be resolving the promise from the loader `defer()`.

### Actual Behavior

After clicking "Update time" button, the previous `useAsyncValue` loader data persists before being replaced by the new loader data after 2 seconds, without showing the "Loading..." text.

I expected that re-running the loader with defer (by changing the search parameters) should always show the `React.Suspense` fallback loading state.

**Comment by brophdawg11** [comment]
This is expected behavior at the moment - please see https://reactrouter.com/en/main/guides/deferred#when-does-the-suspense-fallback-render for an explanation.  

Also related to https://github.com/remix-run/react-router/issues/10638.

The tl;dr; is that search params changing _revalidates_ the current loader data - and `defer` fallbacks only apply to the initial load.  In the future we hope to make this more user-configurable, likely once the new `React.use` hook stabilizes.

**Comment by szemyd** [comment]
Hi! I have the same issue -- is there are recommended workaround? Using `useNavigation().state` instead of `Suspense`?

**Comment by brophdawg11** [comment]
Yeah - `navigation.state` is the workaround if using `defer`.  

Also worth noting that if you are building a SPA then you can actually skip `defer` and just return a raw `Promise` in your JSON object and it won't cause RR to wait on promise resolution on search param updates (this works because the data never has to be serialized over the network like it does in Remix).

```js
function loader() {
  return {
    dataPromise: getData()
  };
}
```

**Comment by brophdawg11** [comment]
You can still use `Await`.  `defer` is what makes it wait for resolution on revalidations.  Remove `defer` and the router will no longer wait for the promises to resolve.  Then you can manage the fallback re-showing or not via `<Suspense key>`

**Comment by julienw** [comment]
hey @brophdawg11 thanks for your answer. I still can't make it work reliably.
I removed `defer` and I can get the fallback to render _sometimes_. Do you know if Suspense waits for some timeout before showing the fallback by default?
You mentioned using `key` to force Suspense to reset its state but when I tried this didn't make a difference. By chance would you have an example that I could study?
Thanks again

**Comment by brophdawg11** [comment]
I don't have an example handy - I would suggest asking in the Remix discord https://rmx.as/discord

**Comment by julienw** [comment]
I ended up returning a `generation` value from the loader: this value is incremented each time the loader runs. Then I can use this value as the `key` in Suspense.
Using a simple key such as `Date.now()` wasn't working well mostly because of the spurious rerender that I filed in https://github.com/remix-run/react-router/issues/11864.
